strax/processing/hitlets.py

```diff
@@ -0,0 +1,526 @@
+                res = buffer[offset]
+                res['time'] = lhc['time']
+                res['length'] = (lhc['endtime'] - lhc['time'])//lhc['dt']
+                res['channel'] = lhc['channel']
+                res['record_i'] = lhc['record_i']
+                res['dt'] = lhc['dt']
+                offset += 1
+                if offset == len(buffer):
+                    yield offset
+                    offset = 0
+
+                # Updating current last hit:
+                lhc['time'] = st
+                lhc['endtime'] = et
+                lhc['channel'] = hc
+                lhc['record_i'] = r_i
+
+    # We went through so now we have to save all remaining hits:
+    mask = last_hit_in_channel['time'] != 0
+    for lhc in last_hit_in_channel[mask]:
+        res = buffer[offset]
+        res['time'] = lhc['time']
+        res['channel'] = lhc['channel']
+        res['length'] = (lhc['endtime'] - lhc['time'])//lhc['dt']
+        res['record_i'] = lhc['record_i']
+        res['dt'] = lhc['dt']
+        offset += 1
+        if offset == len(buffer):
+            yield offset
+            offset = 0
+    yield offset
+
+
+@export
+@numba.njit(nogil=True, cache=True)
+def refresh_hit_to_hitlets(hits, hitlets):
+    """
+    Function which copies basic hit information into a new hitlet array.
+    """
+    nhits = len(hits)
+    for ind in range(nhits):
+        h_new = hitlets[ind]
+        h_old = hits[ind]
+
+        h_new['time'] = h_old['time']
+        h_new['length'] = h_old['length']
+        h_new['channel'] = h_old['channel']
+        h_new['area'] = h_old['area']
+        h_new['record_i'] = h_old['record_i']
+        h_new['dt'] = h_old['dt']
+
+
+@export
+@numba.njit(nogil=True, cache=True)
+def get_hitlets_data(hitlets, records, to_pe):
+    """
+    Function which searches for every hitlet in a given chunk the 
+    corresponding records data.
+    
+    :param hitlets: Hitlets found in a chunk of records.
+    :param records: Records of the chunk.
+    :param to_pe: Array with area conversion factors from adc/sample to 
+        pe/sample
+    
+    Note:
+        hitlets must have a "data" and "area" field.
+    
+    The function updates the hitlet fields time, length (if necessary 
+    e.g. hit was extended in regions of now records) and area
+    according to the found data.
+    """
+
+    rlink = strax.record_links(records)
+    for h in hitlets:
+        data, start_time = get_single_hitlet_data(h, records, *rlink)
+        h['length'] = len(data)
+        h['data'][:len(data)] = data * to_pe[h['channel']]
+        h['time'] = start_time
+        h['area'] = np.sum(data * to_pe[h['channel']])
+
+
+TRIAL_COUNTER_NEIGHBORING_RECORDS = 100
+@export
+@numba.njit(nogil=True, cache=True)
+def get_single_hitlet_data(hitlet, records, prev_r, next_r):
+    """
+    Function which gets the data of a single hit or hitlet according
+    to the specified time and length of the object.
```

Comment:
I'm not sure if I follow the docstring, especially given the arguments

Reply:
Okay I will change it. My intention was to say something like: For hits you will only get the central part without LE/RE